pkg/deploy/broker.go

```diff
@@ -123,47 +122,45 @@ func Broker(options *BrokerOptions, restConfigProducer restconfig.Producer) erro
 	}
 
 	err = subctlData.WriteToFile(brokerDetailsFilename)
-	status.End(cli.CheckForError(err))
+	reporter.End(cli.CheckForError(err))
```

Comment:
I think `reporter` should be notified of the error message via `Failure`.  We also return the error to the caller. Since the caller passes in the `reporter`, we should let the caller end it. So here we just return the error. The caller woild do:

```
reporter := ...
err := deploy.Broker(..., reporter)
if err != nil {
    reporter.Failure("Error deploying Broker: %v", err)
}

reporter.End()
```

```suggestion
```